### PR TITLE
Uses correct version for automatic upshift, sets coordinate Raft log entries to ignore.

### PR DESCRIPTION
**Pull request by slackpad**
This should fix https://github.com/hashicorp/consul/issues/1346.

### CODE REVIEW

File: consul/util.go [hunk 1]
```diff
@@ -95,6 +95,30 @@ func ensurePath(path string, dir bool) error {
...
+		}
+		numServers++
+
+		vsn_str := m.Tags["vsn_max"]
+		vsn, err := strconv.Atoi(vsn_str)
+		if err != nil {
+			return false, err
+		}
+
+		if vsn >= int(version) {
```
Comment (ryanuber) [inline consul/util.go]: Should there also be a `<=` here for the `vsn_min`? Likely wouldn't see that any time soon I guess.

**Comment by ryanuber** [comment]
Just one comment (more of a question), but LGTM!

File: consul/util.go (hunk 1 above)
Comment (slackpad) [inline consul/util.go]: Good call. It's more important to have old clients do this right than newer ones so I'll add the full interval check.

**Comment by slackpad** [comment]
Build fails are transient and not related to these changes.

**Pull request merged**